zc_install/includes/nginx_conf/ngx_conf_server.txt

```diff
@@ -0,0 +1,349 @@
+	location ^~ %%http_store_folder%%/cache {
+		internal;
+	}
+
+
+###############################
+# Zencart 'logs' Folder
+###############################	
+## Allow and fastrack internal requests to 'logs'
+## Returns '404 Not Found' http code for external requests
+	location ^~ %%http_store_folder%%/logs {
+		internal;
+	}
+
+
+###############################
+# Zencart ADMIN Folder
+###############################	
+## Allow all 'POST', 'GET' and 'HEAD' requests to all PHP files
+## More secure to specifically allow eligible files but not practical
+	location ~ %%http_store_folder%%/%%admin_folder%%/(.+)\.php$ {
+		if ( $request_method !~ ^(GET|HEAD|POST)$ ) {
+			return 403;
+		}
+		error_page 418 = @zencart_php_handler;
+		return 418;
+	}
+## Allow all 'GET' and 'HEAD' requests to specific static file types
+	location ~ %%http_store_folder%%/%%admin_folder%%(.*)$ {
+		if ( $request_method !~ ^(GET|HEAD)$ ) {
+			return 403;
+		}
+		location ~ %%http_store_folder%%/%%admin_folder%%/(.+)\.(ico|jpe?g|gif|otf|woff2?|ttf|webp|png|swf|flv|pdf|svg?z)$ {
+			add_header ETag '';
+			add_header Pragma '';
+			add_header Last-Modified '';
+			add_header Cache-Control 'max-age=864000, public, must-revalidate';
+			
+			# Optional caching improvement
+			# Uncomment if accompanying 'map' block in HTTP section has been added
+			#expires $zencart_expires;		
+		}
+		location ~ %%http_store_folder%%/%%admin_folder%%/(.+)\.(html|htm|xml|txt|xsl)$ {
+			add_header ETag '';
+			add_header Pragma '';
+			add_header Cache-Control 'max-age=7200, must-revalidate';
+			
+			# Optional caching improvement
+			# Uncomment if accompanying 'map' block in HTTP section has been added
+			#expires $zencart_expires;		
+		}
+		location ~ %%http_store_folder%%/%%admin_folder%%/(.+)\.(js|css)$ {
+			# Optional caching improvement
+			# Uncomment if accompanying 'map' block in HTTP section has been added
+			#expires $zencart_expires;		
+		}
+		# Redirect all requests to ADMIN root to ADMIN/index.php 
+		location ~ %%http_store_folder%%/%%admin_folder%%(/?)$ {
+			return %%http_store_folder%%/%%admin_folder%%/index.php;
+		}
+		# Forbid any other request to '%%admin_folder%%' and any subfolders
+		return 403;
+	}
+
+
+###############################
+# Zencart ROOT Folder
+###############################	
+## Allow all requests to specific static file types
+	location ~ %%http_store_folder%%/(.+)\.(md|ico|txt|html)$ {
+		#expires $zencart_expires;
+	}
+## Allow all requests to specific PHP files directly under the Zencart ROOT folder
+	location ~ %%http_store_folder%%/(ajax|index|page_not_found|ipn_main_handler)\.php$ {
+		error_page 418 = @zencart_php_handler;
+		return 418;
+	}
+
+	
+###############################
+# Zencart Catchall
+###############################	
+## Redirect any request not previously explicitly handled to main Zencart index.php file
+	location ~ %%http_store_folder%%(.*)$ {
+		return %%http_store_folder%%/index.php;
```

Comment:
This `return` is coming back as invalid.

But changing this return line to a `try` seems to work: `try_files $uri $uri/ /index.php?$args ;`  (and similarly with the one for admin/index.php 26 lines earlier).